### ISSUE TITLE
(typescript) each collection is missing generic type

### ISSUE THREAD

**Issue by sqal** [kenwheeler/cash#289]
### Current behavior

```js
$('[data-title]').each((_index, el: HTMLElement) => {});
```

raises an error

```
(parameter) el: Element
Argument of type '(_index: number, el: Element) => void' is not assignable to parameter of type 'EachCallback<Ele>'.
  Types of parameters 'el' and 'ele' are incompatible.
    Type 'Ele' is not assignable to type 'Element'.
      Type 'Node' is missing the following properties from type 'Element': assignedSlot, attributes, classList, className, and 72 more.ts(2345)
```

### Expected behavior

No error :)

### Additional info:

I noticed that in declaration file there are 2 interface that describe `each` function and the second one is missing declaration of `Ele` generic type, which is causing described issue.

```typescript
interface CashStatic {
  each<T>(arr: ArrayLike<T>, callback: EachCallback<T>): void;
}
interface Cash {
  each(callback: EachCallback<Ele>): this;
}
```

This code fixes it for me:

```typescript
interface Cash {
  each<Ele>(callback: EachCallback<Ele>): this;
}
```

**Comment by fabiospampinato** [comment]
I first made the codebase much more strict, but this was pretty annoying as now we had to always consider the cases where `window` or `document` might be in the collection, when realistically we basically never put those into our collections by mistake, so I then made the codebase a bit more loose in this regard.

But I think I forgot to make some methods loose too.

Your solution I think "fixes" that because now when you don't provide a type that "Ele" type is "any" by default, and doesn't refer to the "Ele" type we are using throughout the codebase.

I'll get back to this later this week probably, thanks for reporting the issue 👍
